x-pack/plugins/maps/public/components/metrics_editor/metric_editor.tsx

```diff
@@ -57,30 +57,39 @@ export function MetricEditor({
     if (!metricAggregationType) {
       return;
     }
-    const newMetricProps = {
-      ...metric,
-      type: metricAggregationType,
-    };
 
     // unset field when new agg type does not support currently selected field.
-    if (metric.field && metricAggregationType !== AGG_TYPE.COUNT) {
+    if ('field' in metric && metric.field && metricAggregationType !== AGG_TYPE.COUNT) {
       const fieldsForNewAggType = filterFieldsForAgg(fields, metricAggregationType);
       const found = fieldsForNewAggType.find((field) => {
         return field.name === metric.field;
       });
-      if (!found) {
-        newMetricProps.field = undefined;
+      if (found) {
```

Comment:
Not sure all of these cases are needed. There should be a cleaner way to do this. You are just removing `field` key. This can be done with some casting instead of all these if/else/else nesting.

Reply:
+1 that this christmas tree is unreadable. What do you think of new version?